src/Composer/ComposerWidget.vala

```diff
@@ -405,6 +415,34 @@ public class Mail.ComposerWidget : Gtk.Grid {
         }
     }
 
+    private void on_add_attachment () {
+        var filechooser = new Gtk.FileChooserNative (
+            _("Choose a file"),
+            (Gtk.Window) get_toplevel (),
+            Gtk.FileChooserAction.OPEN,
+            _("Attach"),
+            _("Cancel")
+        );
+
+        if (filechooser.run () == Gtk.ResponseType.ACCEPT) {
+            filechooser.hide ();
+            foreach (File file in filechooser.get_files ()) {
+                try {
+                    var info = file.query_info ("standard::*", 0);
```

Comment:
```suggestion
                    var info = file.query_info ("standard::*", 0);
```
I'd prefer that step to be inside the `Attachment` object

Reply:
That way we can query less properties than `"standard::*"` which means a lot of things